Modules/Core/Common/include/itkMacro.h

```diff
@@ -157,29 +140,39 @@ namespace itk
  * MSVC++ 6.0 _MSC_VER = 1200
  * MSVC++ 5.0 _MSC_VER = 1100
 */
-#if defined( _MSC_VER ) && ( _MSC_VER < 1310 )
-//#error "_MSC_VER < 1310 (MSVC++ 7.1) not supported under ITKv4"
+#if defined( _MSC_VER ) && ( _MSC_VER < 1600 )
+  #error "MSVC++ < 10.0 is not supported under ITKv5"
 #endif
 #if defined( __SUNPRO_CC ) && ( __SUNPRO_CC < 0x590 )
-#error "__SUNPRO_CC < 0x590 not supported under ITKv4"
+  #error "SUNPro C++ < 0x590 is not supported under ITKv4 and above"
```

Comment:
It looks like we should check for version `0x5140` here. https://github.com/Tencent/rapidjson/issues/1308 and [12.4 docs](https://docs.oracle.com/cd/E37069_01/html/E37071/gncix.html).